### DISCUSSION TITLE
Is there a way to tell the DFA scanner to ignore a certain line of code?

### DISCUSSION THREAD

**Discussion by mattlacey** [forcedotcom/sfdx-scanner#1655] [Q&A] [answered]
I've got a date variable that's initialised like so:

DateTime sevenDaysAgo = System.now().addDays(-7);

This is then used in a where clause:

` AND CreatedDate > :sevenDaysAgo `

The scanner flags this as a problem: "Null check is missing for variable sevenDaysAgo used in SOQL query."

Of course I can, and will swap this out for `LAST_N_DAYS` in the SOQL, but I feel ike there should be a way to disable this false positive as there are no other statements using that variable in-between an if it is somehow `null` I suspect the system has bigger issues :) Is there a way to tell the scanner to ignore this? Granted I have a workaround in this case but if I didn't it'd be silly littering the code with a check that doesn't need to be there.

**Comment by johnbelo** [comment, marked as answer]
Yes see here: https://developer.salesforce.com/docs/platform/salesforce-code-analyzer/guide/working-with-graph-engine.html (under "Add Engine Directives")

**Comment by mattlacey** [comment]
Ah thanks, will give that a go.
